addons/website_livechat/static/src/models/visitor/visitor.js

```diff
@@ -0,0 +1,143 @@
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        static convertData(data) {
+            const data2 = {};
+            if ('country_id' in data && data.country_id) {
+                data2.country = [['insert', {
+                    id: data.country_id,
+                    code: data.country_code,
+                }]];
+            }
+            if ('history' in data) {
+                data2.history = data.history;
+            }
+            if ('is_connected' in data) {
+                data2.is_connected = data.is_connected;
+            }
+            if ('lang' in data) {
+                data2.lang = data.lang;
+            }
+            if ('name' in data) {
+                data2.name = data.name;
+            }
+            if ('partner_id' in data) {
+                if (data.partner_id) {
+                    data2.partner = [['insert', { id: data.partner_id }]];
+                } else {
+                    data2.partner = [['unlink-all']];
+                }
+            }
+            if ('website' in data) {
+                data2.website = data.website;
+            }
+            return data2;
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeAvatarUrl() {
+            if (!this.partner) {
+                return '/mail/static/src/img/smiley/avatar.jpg';
+            }
+            return this.partner.avatarUrl;
+        }
+
+        /**
+         * @private
+         * @returns {mail.country}
+         */
+        _computeCountry() {
+            if (this.partner && this.partner.country) {
+                return [['link', this.partner.country]];
+            }
+            if (this.country) {
+                return [['link', this.country]];
+            }
+            return [['unlink']];
+        }
+
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeNameOrDisplayName() {
+            if (this.partner) {
+                return this.partner.nameOrDisplayName;
+            }
+            return this.name;
+        }
+    }
+
+    Visitor.fields = {
+        avatarUrl: attr({
+            compute: '_computeAvatarUrl',
+            dependencies: [
+                'partner',
+                'partnerAvatarUrl'
+            ]
+        }),
+        country: many2one('mail.country', {
+            compute: '_computeCountry',
+            dependencies: [
+                'partnerCountry',
+            ]
+        }),
+        history: attr(),
+        is_connected: attr(),
+        /**
+         * Name of the language of the visitor.
+         */
```

Comment:
Provide an example (_"e.g. "English""_), clarify it's locale-dependent.

It's not that frequent to speak with _"name of the language"_. Even if it's technically correct, suggestions above ease readability by a lot IMO.